---
title: Data Science Fellowship - Rotation 2
author: Kyle Webb
date: '2019-10-29'
slug: data-science-fellowship-rotation-2
categories:
  - data science
tags:
  - data science
subtitle: ''
description: 'TRIAGE - Network Visualization'
image: ''
draft: true
---

In my second rotation as a data science fellow for the NIH (see [here]("/2019/03/21/data-science-fellowship-rotation-1") for more details), I worked with the Laboratory of Immune System Biology (LISB) in the Division of Intramural Research (DIR) at the National Institute of Allergy and Infectious Diseases (NIAID). One of the major goals of this rotation was to improve the network visualization capabilities of a bioinformatics screening tool named TRIAGE (**T**hroughput **R**anking by **I**terative **A**nalysis of **G**enomic **E**nrichment). TRIAGE is a tool that facilitates hypothesis generation for genome-wide perturbation studies like those designed with short interfering RNA (siRNA), RNA interference (RNAi), and CRISPR/Cas9 assays.

After user defined inputs and [KEGG](https://www.genome.jp/kegg/kegg1.html) and [STRING](https://string-db.org/) enrichment, TRIAGE provides the *hit genes* from the analysis. These *hit genes* can be downloaded by the user or, using the network functionality of the TRIAGE application, the user can investigate which genes correspond to which and highlight the genetic pathways that connect them. Before I came on to the project, this is what a typical network graph looked like:

<!--- place TRIAGE network graph here --->

The network graph above shows each individual node as a gene from the output of TRIAGE and the connected edges between them are biologically driven connections annotated in STRING. Though this network graph was certainly useful, it did have some drawbacks:
- No user-operability and commands such as freezing one or multiple connections
- Not subsetting genes involved in multiple pathways
- No ability to click through pathways and log the path in a dataset or in the visualization itself

I was able to build these changes and many others into the TRIAGE network visualization by starting from the original code of the *networkD3* R package and rewriting customized functions using D3.js. The D3 network files that I wrote were then connected back to the TRIAGE app using the *r2d3* to help with the cross talk between these two languages. Here's an image of the new network visualization portion of the app. 

<!--- other TRIAGE image --->

Try TRIAGE out by downloading [this dataset]() on a study....
Next, visit the [TRIAGE website](), hosted by NIAID. Input the downloaded data from the ___ study and investigate potential pathways. 

*This research was supported in part by an appointment to the National lnstitute of Allergy and lnfectious Diseases (NIAID) Emerging Leaders in Data Science Research Participation Program. This program is administered by the Oak Ridge lnstitute for Science and Education through an interagency agreement between the U.S. Department of Energy and the National lnstitutes of Health.*
